x/merkledb/proof_test.go

```diff
@@ -1511,3 +1505,70 @@ func TestChangeProofUnmarshalProtoInvalidMaybe(t *testing.T) {
 	err := proof.UnmarshalProto(protoProof)
 	require.ErrorIs(t, err, ErrInvalidMaybe)
 }
+
+func TestProofProtoMarshalUnmarshal(t *testing.T) {
+	require := require.New(t)
+	rand := rand.New(rand.NewSource(1337)) // #nosec G404
```

Comment:
Instead of using RNG i would prefer if we just had pre-defined test cases because I feel like they're easier to debug when they break (even if this is deterministic because of seeding). Alternatively doing something similar to what Alberto has been doing with property tests might be another idea worth considering